frontend/src/components/utils/Authentication.tsx

```diff
@@ -0,0 +1,56 @@
+import axios from 'axios';
+import { useEffect } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+
+const authenticatedRouts = ['/home'];
+const unauthenticatedRouts = ['/login', '/registration'];
+
+const Authentication = ({children}: any) => {
+    const history = useHistory();
+    const location = useLocation();
+
+    useEffect(() => {
+        let isAuthenticated = false;
```

Comment:
Save `isAuthenticated` in `useState` hook, you will need this info in other parts of application.